R/LSgameofchance.R

```diff
@@ -15,272 +15,146 @@
+      which(xPoints == max(xPoints))
+    ))
   
+  if(sum(c(xPoints, probWin) > 0) != length(c(xPoints, probWin)))
+    .quitAnalysis(gettext("Warning: No negative input values! Adjust the inputs!"))
   
-
   
   ## Summary Table
   summaryTable <- createJaspTable(title = gettext("Summary Table"))
   
-  summaryTable$dependOn(c("n", "k", "t", "p", "s", "check"))
+  summaryTable$dependOn(c("players", "winPoints", "nSims", "CI"))
   summaryTable$addCitation("JASP Team (2018). JASP (Version 0.9.2) [Computer software].")
   
-  summaryTable$addColumnInfo(name = "players",  title = gettext("Players"),          type = "string")
-  summaryTable$addColumnInfo(name = "pPoint",   title = gettext("Pr(win a point)"),  type = "number")
-  summaryTable$addColumnInfo(name = "pA",       title = gettext("Analytical"),       type = "number", 
-                             overtitle = gettext("Pr(win the game)"))
-  summaryTable$addColumnInfo(name = "pS",       title = gettext("Simulated"),         type = "number", 
-                             overtitle = gettext("Pr(win the game)"))
+  summaryTable$addColumnInfo(name = "players",   title = gettext("Players"),   type = "string")
+  summaryTable$addColumnInfo(name = "pPoint",       title = gettext("p(win 1 point)"), type = "number")
+  summaryTable$addColumnInfo(name = "pointsGained", title = gettext("Points Gained"),   type = "integer")
+  summaryTable$addColumnInfo(name = "pA",   title = gettext("Analytical"),   type = "number", 
+                             overtitle = gettext("p(win the game)"))
+  summaryTable$addColumnInfo(name = "pS",   title = gettext("Simulated"),   type = "number", 
+                             overtitle = gettext("p(win the game)"))
+  
+  # add footnote for normalization of prob
+  # if(sum(as.numeric(unlist(strsplit(probWin,",")))) != 1){
+  #   summaryTable$addFootnote("The probability entered does not sum to 1. Already normalized!")
+  # }
   
   ## Credible Interval Plot
-  CIPlot <- createJaspPlot(title = gettext("Probability of Player 1 Winning"),  width = 480, height = 320)
-  CIPlot$dependOn(c("n", "k", "t", "p", "s", "check"))
+  CIPlot <- createJaspPlot(title = "Probability of Player 1 Winning",  width = 480, height = 320)
+  CIPlot$dependOn(c("players", "winPoints", "nSims", "CI"))
   CIPlot$addCitation("JASP Team (2018). JASP (Version 0.9.2) [Computer software].")
   
   # column specification
   CIPlot0 <- ggplot2::ggplot(data= NULL) + 
-    ggplot2::xlab(gettext("Number of Simulated Games")) + 
-    ggplot2::ylab(gettext("Pr(Winning the Game)")) +
-    ggplot2::coord_cartesian(xlim = c(0, input$s), ylim = c(0, 1)) 
+    #ggplot2::ggtitle("Probability of Player 1 Winning") +
+    ggplot2::xlab("Number of Simulated Games") + 
+    ggplot2::ylab("Pr(Winning the Game)") +
+    ggplot2::coord_cartesian(xlim = c(0, nSims), ylim = c(0, 1)) 
   
   ## fill in the table and the plot 
-  if (input$n == 2 & max(k) < input$t){
+  if (nPlayers == 2 & max(xPoints) < winPoints){
```

Comment:
Is it possible to write this in a general form for any number of players, instead of splitting it for `nPlayers == 2` and `nPlayers > 2`?

Reply:
The negative multinomial distribution cannot cover the negative binomial distribution, so I think separating this two conditions is the right thing to do.